### DISCUSSION TITLE
Send-PnPMail -To

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/powershell#972] [Q&A]
for some reason I can't figure out how to create `-To <String[]>` from an array I make from the members of  Microsoft365Group

```
$toAddress = [string[]]::new($members.Length)
for ($i = 0; $i -lt $members.Length; $i++) {
    $toAddress[$i] = $members[$i].Email
}
```
sends email only to the first address in the array

Also I can't get the body to be multi line

I add  `n at the end of each sentence but it does not do it,
I also tried "&lt;br/&gt;" (with the &xx;) at the end but it just shows "&lt;br/&gt;"

**Comment by LeonArmston** [comment]
Hi Ofer

Hope this helps below - you need to join the $members.Email array into one string using -Join "," then I have showed you how to use HTML in emails so you can use new line and any other html formatting you like:

```powershell

$email = @"
<!DOCTYPE html>
<html>
<body>

<h1>My Email</h1>
<p>My first paragraph.<br />
New line here.</p>

</body>
</html>
"@ 

$group = Get-PnPMicrosoft365Group -Identity "Your Group"

$members = Get-PnPMicrosoft365GroupMembers -Identity $group

Send-PnPMail -To ($members.Email -join ",") -Subject "Test Email" -Body $email 
```

Thanks

**Reply by Ofer-Gal** [reply]
Thank you much
I get The e-mail message cannot be sent. Make sure the e-mail has a valid recipient.
I am using 1.7.0      PnP.PowerShell
![image](https://user-images.githubusercontent.com/5054931/128066407-51a70b3c-f1a8-4218-807d-30443f2dacac.png)
The array has 1 item rather than 3 
So I tried with -To $members.Email and that is a [string[3]]
![image](https://user-images.githubusercontent.com/5054931/128070021-377521c8-1762-4ba1-b1a3-c2759cc9fd45.png)
Send-PnPMail IS OK  but only I (OGal@ecgmc.com) get an email (Very nicely laid out Thank you much :-))
Could this be because I authenticate with an app only (app ID and certificate)?

**Reply by Ofer-Gal** [reply]
I tried to send the emails one by one and saw that the issue is with any address that is not mine.
The error is:
The e-mail message cannot be sent. Make sure the e-mail has a valid recipient.
If I add -From "{Some email}" -Password "{password}" there is no error but no email arrives

**Reply by LeonArmston** [reply]
Hi

Your first screenshot s how the email string needs to appear for multiple users: "user@domain.com,user@domain1.com"

I get the same error message "The e-mail message cannot be sent. Make sure the e-mail has a valid recipient." & I think it sees internal email addresses (same M365 tenant) as external so rejects them (might be a bug). If I add -From "{Some email}" -Password "{password}"  the email is sent fine to six recipients

![image](https://user-images.githubusercontent.com/12968962/128147640-c465a928-6860-4523-b06c-194b65ad656d.png)

**Reply by Ofer-Gal** [reply]
In my DEV tenant it works when -From "{Some email}" -Password "{password}" is my credentials But Not in the production Tenant that has MFA. 
In any case using My password in a script is not a good practice. 
Guess I will have to write an Azure Function instead of this.
Thank you
